database/db_dispatch.cpp

```diff
@@ -164,6 +165,66 @@ db_dispatch::handle_swarm_response(const bzn_envelope& response)
     return false;
 }
 
+bool
+db_dispatch::handle_swarm_error(const bzn_envelope& response)
+{
+    swarm_error err;
+    if (!err.ParseFromString(response.swarm_error()))
+    {
+        LOG(error) << "Failed to parse error response from swarm: " << response.DebugString().substr(0, MAX_MESSAGE_SIZE);
+        return true;
+    }
+
+    uint64_t nonce{0};
+    try
+    {
+        nonce = std::stoull(err.data());
+    }
+    catch(std::exception& e)
+    {}
+
+    if (!nonce)
+    {
+        LOG(error) << "Unable to extract nonce from swarm error";
+        return true;
+    }
+
+    auto i = this->messages.find(nonce);
+    if (i == this->messages.end())
+    {
+        LOG(trace) << "Ignoring error response for unknown or already processed message: " << nonce;
+        return false;
+    }
+    auto& info = i->second;
```

Comment:
do you need this way up here? It's not used until #221

Reply:
done